pandas/tests/series/indexing/test_where.py

```diff
@@ -452,3 +452,18 @@ def test_where_empty_series_and_empty_cond_having_non_bool_dtypes():
     ser = Series([], dtype=float)
     result = ser.where([])
     tm.assert_series_equal(result, ser)
+
+
+def test_where_category_not_dropping():
+
+    s = pd.Series(["A", "A", "B", "B", "C"], dtype="category")
+    exp = s.dtype
+    s.where(s != "C", s)
+    res = s.dtype
+    assert exp == res
+
+
+def test_where_categorical_frame_ValueError():
+    s = pd.Series(["A", "A", "B", "B", "C"], dtype="category")
+    df = s.to_frame()
+    df.where(df != "C")
```

Comment:
here also construct the expected frame and `assert_frame_equal` against that

Reply:
But here, after we apply df.where(df != "C") the result frame would have a NaN category. How can I construct the expected then?

Reply:
You can use `np.nan` or `pd.NA` to enter the null values.

We have a few examples in the user guide, under `Object Creation`:
https://pandas.pydata.org/pandas-docs/stable/user_guide/categorical.html

The `pd.Categorical` API reference is another place to look:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Categorical.html

Reply:
Great, thanks!